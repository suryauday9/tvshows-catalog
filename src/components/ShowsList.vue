<template>
    <v-container grid-list-lg>
        <v-layout row wrap> 
            <v-sheet
                class="mx-auto"
                max-width="100%"
            >
                <v-slide-group
                    multiple
                    show-arrows
                >
                    <v-slide-item
                        v-for="genre in genres"
                        :key="'genre-' + genre"
                        class="pr-4"
                        v-slot="{ active, toggle }"
                    >
                        <v-btn
                        class="mx-2"
                        :input-value="active"
                        active-class="purple white--text"
                        depressed
                        rounded
                        @click="toggle"
                        >
                        {{ genre }}
                        </v-btn>
                    </v-slide-item>
                </v-slide-group>
            </v-sheet>
        </v-layout>
    </v-container>
</template>

<script>
export default {
    name: 'ShowsList',
    props: {
        genresData: Array,
    },
    data () {
        return {
            genres: this.genresData ? this.genresData : [],      
        }
    }
};
</script>

<style scoped>
ul,
ol {
  padding-left: 24px;
}
.title {
  font-size: 20px !important;
  font-weight: 800;
  line-height: 1 !important;
  letter-spacing: 0.02em !important;
  font-family: 'Roboto', sans-serif !important;
}
.subheading {
  font-size: 16px !important;
  font-weight: 500;
  margin: 6px 9px;
  text-align: left;
}
.recipe-details {
    margin: 3px 12px;
}
.recipe-desc {
    text-align: left;
    padding: 6px 9px;
}
ul.items-list {
    text-align: left;
}
.videoframe {
    width: 96%;
    height: 270px;
}
</style>
