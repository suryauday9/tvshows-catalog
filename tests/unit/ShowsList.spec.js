import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';

import { initializeApp } from './initializeApp';
import ShowsList from '../../src/components/ShowsList.vue';
import { genres } from './response';

const vuetify = new Vuetify();
const localVue = initializeApp(createLocalVue());

describe('PageHeader component', () => {  
    let wrapper;
    // const router = new VueRouter(routes);
    
    // const store = new Vuex.Store({
    //     state: {
    //         shows: showsData,
    //         genres: [],
    //         showDetails: [],
    //         showThumbnails: [],
    //         showSearchbar: false
    //     },
    //     actions: {
    //         getShowsBySearch: jest.fn(),
    //         getShowslist: jest.fn()
    //     }
    // });

    const mountComponent = (options) => {
        return shallowMount(ShowsList, {
            localVue,
            // router,
            // store,
            propsData: { 'genresData' : genres },
            // data() {
            //     return {
            //         genres: this.propsData
            //     }
            // },
            vuetify,
            ...options,
        })
    }

    beforeEach(() => {
        wrapper = mountComponent();
    })

    afterEach(() => {
        wrapper.destroy();
    })

    it('renders component correctly', () => {
        // see if component renders
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should be a Vue instance', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    it('rendered component data should match with store', () => {
        // see if data loads
        expect(wrapper.vm.genres).toEqual(genres);
    });

    it('should handle empty props without breaking', () => {
        const sltWrapper = mount(ShowsList, {
            localVue,
            vuetify,
            propsData: { },
        });
        expect(sltWrapper.vm.genres).toEqual([]);
    });

});