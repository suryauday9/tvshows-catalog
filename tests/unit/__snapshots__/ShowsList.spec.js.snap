// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component renders component correctly 1`] = `
<v-container-stub
  grid-list-lg=""
  tag="div"
>
  <v-layout-stub
    row=""
    tag="div"
    wrap=""
  >
    <v-sheet-stub
      class="mx-auto"
      maxwidth="100%"
      tag="div"
    >
      <v-slide-group-stub
        activeclass="v-slide-item--active"
        mobilebreakpoint="1264"
        multiple="true"
        nexticon="$next"
        previcon="$prev"
        showarrows="true"
        tag="div"
      >
        <v-slide-item-stub
          class="pr-4"
        />
        <v-slide-item-stub
          class="pr-4"
        />
        <v-slide-item-stub
          class="pr-4"
        />
        <v-slide-item-stub
          class="pr-4"
        />
        <v-slide-item-stub
          class="pr-4"
        />
        <v-slide-item-stub
          class="pr-4"
        />
      </v-slide-group-stub>
    </v-sheet-stub>
  </v-layout-stub>
</v-container-stub>
`;
